File: kdtree.py
import numpy as np
import scipy.spatial as spatial
import time

length_of_video = 1*60*60 # seconds
interest_points = np.float32(np.random.rand(50 * length_of_video * 30, 128))

# start time = current time
start_time = time.time()
# perform kdtree
tree = spatial.cKDTree(interest_points, leafsize=10)
# end time 
end_time = time.time()
print("Time taken to build KDTree: ", end_time - start_time)

File: kdtree.jl
using NearestNeighbors

length_of_video = 1*60*60 # seconds
interest_points = rand(Float32, 128, 50 * length_of_video * 30)

start_t = time()
kdtree = KDTree(interest_points, leafsize=10, reorder=true)
end_t = time()

println("KDTree construction time: ", end_t - start_t, " seconds")